src/components/CowPen/CowPen.js

```diff
@@ -162,15 +163,21 @@ export class Cow extends Component {
   }
 
   componentDidMount() {
+    this.isComponentMounted = true
```

Comment:
is there a reason you didn't put this into state? just curious because my experience with this pattern has always been a state value

Reply:
Unfortunately that wouldn't work here. I tried this:

```diff
diff --git a/src/components/CowPen/CowPen.js b/src/components/CowPen/CowPen.js
index 4d5f3173..8d3407ca 100644
--- a/src/components/CowPen/CowPen.js
+++ b/src/components/CowPen/CowPen.js
@@ -29,12 +29,12 @@ export class Cow extends Component {
     showHugAnimation: false,
     x: randomPosition(),
     y: randomPosition(),
+    isComponentMounted: false,
   }
 
   repositionTimeoutId = null
   animateHugTimeoutId = null
   tweenable = new Tweenable()
-  isComponentMounted = false
 
   static flipAnimationDuration = 1000
   static transitionAnimationDuration = 3000
@@ -163,12 +163,12 @@ export class Cow extends Component {
   }
 
   componentDidMount() {
-    this.isComponentMounted = true
+    this.setState({ isComponentMounted: true })
     this.scheduleMove()
     ;(async () => {
       const cowImage = await getCowImage(this.props.cow)
 
-      if (!this.isComponentMounted) return
+      if (!this.state.isComponentMounted) return
 
       this.setState({ cowImage: cowImage })
     })()
@@ -177,7 +177,7 @@ export class Cow extends Component {
   componentWillUnmount() {
     ;[this.repositionTimeoutId, this.animateHugTimeoutId].forEach(clearTimeout)
 
-    this.isComponentMounted = false
+    this.setState({ isComponentMounted: false })
     this.tweenable.cancel()
   }
 ```


But `this.state.isComponentMounted` wasn't set to `false` in time for it to be evaluated as we expect after the `const cowImage = await getCowImage(this.props.cow)` line.